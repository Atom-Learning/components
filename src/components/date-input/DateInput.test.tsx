import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { DateInput } from '.'

describe("DateInput component", () => {
  it('renders', async () => {
    const { container } = render(<DateInput />)

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(<DateInput />)

    expect(await axe(container)).toHaveNoViolations()
  })
})