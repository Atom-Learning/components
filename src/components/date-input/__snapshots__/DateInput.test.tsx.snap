// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput component renders 1`] = `
@media  {
  .c-fZTBsJ {
    -webkit-appearance: none;
    appearance: none;
    border: 1px solid var(--colors-tonal400);
    border-radius: var(--radii-0);
    box-shadow: none;
    box-sizing: border-box;
    color: var(--colors-tonal600);
    cursor: text;
    display: block;
    font-family: var(--fonts-body);
    padding-left: var(--space-3);
    padding-right: var(--space-3);
    transition: all 100ms ease-out;
    width: 100%;
  }

  .c-fZTBsJ:focus {
    border-color: var(--colors-primary);
    outline: none;
  }

  .c-fZTBsJ[disabled] {
    background-color: var(--colors-tonal100);
    color: var(--colors-tonal400);
    cursor: not-allowed;
  }

  .c-fZTBsJ::placeholder {
    color: var(--colors-tonal300);
    opacity: 1;
  }

  .c-fDzwZw {
    align-items: center;
    -webkit-appearance: none;
    appearance: none;
    background: white;
    border: unset;
    border-radius: var(--radii-0);
    box-sizing: border-box;
    cursor: pointer;
    display: flex;
    justify-content: center;
    padding: unset;
    transition: all 100ms ease-out;
  }

  .c-dbrbZt {
    display: inline-block;
    fill: none;
    stroke: currentcolor;
    stroke-linecap: round;
    stroke-linejoin: round;
    vertical-align: middle;
  }

  .c-kNRrVd {
    background: white;
    border-radius: var(--radii-1);
    box-shadow: var(--shadows-1);
    max-width: 90vw;
    padding: var(--sizes-1);
    position: relative;
  }

@media (prefers-reduced-motion: no-preference) {
    .c-kNRrVd {
      animation-duration: 75ms;
      animation-timing-function: cubic-bezier(0.16, 1, 0.3, 1);
      will-change: transform, opacity;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-kNRrVd[data-state="open"][data-side="top"] {
      animation-name: k-ftcNPK;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-kNRrVd[data-state="open"][data-side="right"] {
      animation-name: k-hbaHED;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-kNRrVd[data-state="open"][data-side="bottom"] {
      animation-name: k-daNevH;
    }
}

@media (prefers-reduced-motion: no-preference) {
    .c-kNRrVd[data-state="open"][data-side="left"] {
      animation-name: k-fvyGIa;
    }
}
}

@media  {
  .c-fZTBsJ-pHZal-size-md {
    height: var(--sizes-4);
    font-size: var(--fontSizes-md);
  }

  .c-fDzwZw-PrXKS-size-lg {
    height: var(--sizes-4);
    width: var(--sizes-4);
  }

  .c-dbrbZt-eAiySG-size-sm {
    height: var(--sizes-1);
    width: var(--sizes-1);
    stroke-width: 1.75;
  }

  .c-kNRrVd-kUaMfZ-size-md {
    max-width: 400px;
  }
}

@media  {
  .c-fDzwZw-otpKd-cv {
    background: transparent;
    color: var(--colors-tonal300);
  }

  .c-fDzwZw-otpKd-cv:not(:disabled):hover,
  .c-fDzwZw-otpKd-cv:not(:disabled):focus {
    color: var(--colors-primaryMid);
  }

  .c-fDzwZw-otpKd-cv:not(:disabled):active {
    color: var(--colors-primaryDark);
  }

  .c-fDzwZw-otpKd-cv[disabled] {
    color: var(--colors-tonal400);
    cursor: not-allowed;
  }
}

@media  {
  .c-PJLV-icmpvrW-css {
    position: relative;
  }

  .c-fDzwZw-ialwQu-css {
    position: absolute;
    top: 0;
    right: 0;
  }

  .c-dbrbZt-ihFPSGE-css {
    height: 20px;
    width: 20px;
  }
}

<div>
  <div
    class="c-PJLV c-PJLV-icmpvrW-css"
  >
    <input
      class="c-fZTBsJ c-fZTBsJ-pHZal-size-md"
      name="date"
      placeholder="DD/MM/YYYY"
      type="text"
      value=""
    />
    <button
      aria-controls="radix-id-7953557446-1"
      aria-expanded="false"
      aria-haspopup="dialog"
      aria-label="Open calendar"
      class="c-fDzwZw c-fDzwZw-PrXKS-size-lg c-fDzwZw-otpKd-cv c-fDzwZw-ialwQu-css"
      data-state="closed"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="c-dbrbZt c-dbrbZt-eAiySG-size-sm c-dbrbZt-ihFPSGE-css"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 5h18v16H3V5zm18 4H3m4-4V3m10 2V3"
        />
        <rect
          height="2"
          width="2"
          x="15"
          y="15"
        />
      </svg>
    </button>
  </div>
</div>
`;
